angular/projects/admin-nrpti/src/app/records/records-resolver.ts

```diff
@@ -43,43 +43,73 @@ export class RecordsResolver implements Resolve<Observable<object>> {
       subset = params.subset.split(',');
     }
 
-
-    const filterParams = {};
+    const and = {};
+    const or = {};
+    const nor = {};
 
     if (params.dateRangeFromFilter) {
-      filterParams['dateRangeFromFilterdateIssued'] = params.dateRangeFromFilter;
+      or['dateRangeFromFilterdateIssued'] = params.dateRangeFromFilter;
     }
 
     if (params.dateRangeToFilter) {
-      filterParams['dateRangeToFilterdateIssued'] = params.dateRangeToFilter;
+      or['dateRangeToFilterdateIssued'] = params.dateRangeToFilter;
     }
 
     if (params.issuedToCompany && params.issuedToIndividual) {
-      filterParams['issuedTo.type'] = 'Company,Individual,IndividualCombined';
+      or['issuedTo.type'] = 'Company,Individual,IndividualCombined';
     } else if (params.issuedToCompany) {
-      filterParams['issuedTo.type'] = 'Company';
+      or['issuedTo.type'] = 'Company';
     } else if (params.issuedToIndividual) {
-      filterParams['issuedTo.type'] = 'Individual,IndividualCombined';
+      or['issuedTo.type'] = 'Individual,IndividualCombined';
     }
 
     if (params.agency) {
-      filterParams['issuingAgency'] = params.agency;
+      or['issuingAgency'] = params.agency;
     }
 
     if (params.act) {
-      filterParams['legislation.act'] = params.act;
+      or['legislation.act'] = params.act;
     }
 
     if (params.regulation) {
-      filterParams['legislation.regulation'] = params.regulation;
+      or['legislation.regulation'] = params.regulation;
     }
 
     if (params.sourceSystemRef) {
-      filterParams['sourceSystemRef'] = params.sourceSystemRef;
+      or['sourceSystemRef'] = params.sourceSystemRef;
     }
 
     if (params.hasDocuments) {
-      filterParams['hasDocuments'] = params.hasDocuments;
+      or['hasDocuments'] = params.hasDocuments;
+    }
+
+    if (params.projects) {
+      const projectNames = [];
+
+      if (params.projects.includes('lngCanada')) {
+        projectNames.push('LNG Canada');
+      }
+
+      if (params.projects.includes('coastalGaslink')) {
+        projectNames.push('Coastal Gaslink');
+      }
+
+      if (params.projects.includes('otherProjects')) {
+        if (projectNames.length === 0) {
+          // Selecting only Other should return all projects EXCEPT for LNG Canada and Coastal Gaslink
+          nor['projectName'] = 'LNG Canada,Coastal Gaslink';
+        } else if (projectNames.length === 1) {
+          if (projectNames[0] === 'LNG Canada') {
+            // Other + LNG Canada is equivalent to NOT Coastal Gaslink
+            nor['projectName'] = 'Coastal Gaslink';
+          } else {
+            // Other + Coastal Gaslink is equivalent to NOT LNG Canada
+            nor['projectName'] = 'LNG Canada';
+          }
+        }
+      } else if (projectNames.length) {
+        or['projectName'] = projectNames.join(',');
+      }
```

Comment:
This is a string lookup, which is not indexed and will be slow as a result.  Instead, search on _epicProjectId: ObjectId("588511c4aaecd9001b825604") or ObjectId("5885101caaecd9001b80b41f").

Reply:
Updated!